src/ColinHDev/CPlot/provider/DataProvider.php

```diff
@@ -1299,4 +1312,222 @@ public function savePlotRate(Plot $plot, PlotRate $plotRate) : Generator {
+                                "originZ" => $originData[2],
+                                "mergedX" => $mergeData[0],
+                                "mergedZ" => $mergeData[1]
+                            ];
+                        }
+                    }
+                    break;
+                default:
+                    return; // don't import anything due to invalid data provider
+            }
+            foreach($records as $record) {
+                // validate offline player data
+                $XUID = null;
+                $offlineData = Server::getInstance()->getOfflinePlayerData($record["owner"]);
+                if($offlineData !== null) {
+                    $XUID = $offlineData->getString("LastKnownXUID", null);
+
+                    // register filler player data
+                    yield from $this->updatePlayerData(
+                        null, // doesn't matter what is input at this point. will overwrite on login
+                        $XUID,
+                        $record["owner"]
+                    );
+                }
+
+                /** @var PlayerData|null $playerData */
+                $playerData = yield $this->awaitPlayerDataByData(
+                    null,
+                    $XUID,
+                    $record["owner"]
+                );
+                if (!($playerData instanceof PlayerData)) {
+                    continue;
+                }
+
+                // load world
+                /** @var WorldSettings|false $world */
+                $world = yield $this->awaitWorld($record["level"]);
+                if($world === false)
+                    continue;
+
+                // load plot
+                /** @var Plot|null $plot */
+                $plot = yield $this->awaitPlot($record["level"], (int)$record["x"], (int)$record["z"]);
+                if($plot === null)
+                    continue;
+
+                // claim plot
+                $senderData = new PlotPlayer($playerData, PlotPlayer::STATE_OWNER);
+                $plot->addPlotPlayer($senderData);
+                yield from DataProvider::getInstance()->savePlotPlayer($plot, $senderData);
+
+				// load helpers
+				foreach($record["helpers"] as $playerName) {
+					// validate offline player data
+					$XUID = null;
+					$offlineData = Server::getInstance()->getOfflinePlayerData($playerName);
+					if($offlineData !== null) {
+						$XUID = $offlineData->getString("LastKnownXUID", "");
+
+						// register player data
+						yield from $this->updatePlayerData(
+							null, // doesn't matter what is input at this point. will overwrite on login
+							$XUID,
+							$playerName
+						);
+					}
+
+					$playerData = yield $this->awaitPlayerDataByData(
+						null,
+						$XUID,
+						$playerName
+					);
+					if (!($playerData instanceof PlayerData))
+						continue;
+
+					$senderData = new PlotPlayer($playerData, PlotPlayer::STATE_HELPER);
```

Comment:
Iirc, plot helpers in MyPlot are always able to build on the plot, while in CPlot helpers are only able to do that if a plot owner is online. So the matching state by behaviour would be `Trusted`.